### PR TITLE
feat: support keyless verification for verify-blob-attestation

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

* feat: supports keyless verification for verify-blob-attestation. this includes loading from a bundle & cert verification options
* cleanup: removes DSSE support from `verify-blob`

### CODE REVIEW

**Comment by haydentherapper** [comment]
```
    testing.go:1097: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestVerifyBlobCmdWithBundle3198511928\001\attestation.txt: The process cannot access the file because it is being used by another process.
```

A file not being closed?

**Comment by dlorenc** [comment]
> Yeah, but everything is using temp Mkdir, which should cleanup after every test run

I think we've seen bugs with this on Windows...

**Comment by asraa** [comment]
> I think we've seen bugs with this on Windows...

Ah got it, I'll add in manual removals then and see if that'll work ok

**Comment by asraa** [comment]
Nope it was my fault, I found a missing file close.

File: cmd/cosign/cli/verify/verify_blob_attestation.go [hunk 1]
```diff
@@ -16,154 +16,312 @@
...
 	CheckClaims   bool
-	KeyRef        string
 	PredicateType string
+	// TODO: Add policies
 
 	SignaturePath string // Path to the signature
 }
 
 // Exec runs the verification command
 func (c *VerifyBlobAttestationCommand) Exec(ctx context.Context, artifactPath string) error {
```
[the remark this answers is not in the document]
Comment (asraa) [reply cmd/cosign/cli/verify/verify_blob_attestation.go:77]: There's subject claim setting and validating the predicate type. There's not that much I can do here that wouldn't degrade clarity (for simple stuff I think it's better to not obscure behind helpers that work slightly different), except for simplifying some options we can clean up generally across all 4 verification funcs. Stuff like:

- SetFulcioCheckOpts
- SetRekorCheckOpts


over full black-box helpers

File: cmd/cosign/cli/options/verify.go [hunk 2]
```diff
@@ -190,18 +188,41 @@ func (o *VerifyDockerfileOptions) AddFlags(cmd *cobra.Command) {
...
 	cmd.Flags().StringVar(&o.SignaturePath, "signature", "",
 		"path to base64-encoded signature over attestation in DSSE format")
+
+	cmd.Flags().StringVar(&o.BundlePath, "bundle", "",
+		"path to bundle FILE")
+
+	cmd.Flags().BoolVar(&o.CheckClaims, "check-claims", true,
+		"whether to check the claims found")
+
+	cmd.Flags().StringVar(&o.RFC3161TimestampPath, "rfc3161-timestamp", "",
```
Comment (haydentherapper) [inline cmd/cosign/cli/options/verify.go:226]: This exists under the common verify options - https://github.com/sigstore/cosign/blob/631a26fb92347230b423bedfbaac84ef177a6fd0/cmd/cosign/cli/options/verify.go#L22-L26

Comment (asraa) [reply cmd/cosign/cli/options/verify.go:226]: Thank you!! I'll rebase and update

Comment (asraa) [reply cmd/cosign/cli/options/verify.go:226]: Hmmm hold on, it doesn't exist there.

**Review by haydentherapper — APPROVED** [review]
Looks great! Just a failing e2e test

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**